os/fs/smartfs/smartfs_utils.c

```diff
@@ -1771,49 +1767,48 @@ int smartfs_examine_sector(struct smartfs_mountpt_s *fs, char *validsectors, int
 			header = (struct smartfs_chain_header_s *)sectordata;
 			nextsector = SMARTFS_NEXTSECTOR(header);
 
-			if (type == SMARTFS_SECTOR_TYPE_DIR) {
-				if (smart_validatesector(fs->fs_blkdriver, logsector, validsectors) == OK) {
-					(*nusedsectors)++;
-				}
-
+			if (!(node->type == SMARTFS_SECTOR_TYPE_DIR || node->type == SMARTFS_SECTOR_TYPE_FILE)) {
+				ret = -EINVAL;
+				fdbg("Error!! illegal node type %d\n", type);
+				goto errout;
+			}
+			if (smart_validatesector(fs->fs_blkdriver, logsector, validsectors) == OK) {
```

Comment:
Now I think return of smart_validatesector is not ok, it should stop mark and continue next.
So it should sperated both of file & directory case.
1. File : continue;
2. Directory : SMARTFS_NEXTSECTOR

Reply:
Modified as suggested.